src/main/java/com/enterprise/eventmanagementsystem/controller/EventController.java

```diff
@@ -72,15 +73,13 @@ public String save(@ModelAttribute("event") Event event) throws Exception {
      */
     @PostMapping("/createEvent")
     public ResponseEntity createEvent(@RequestBody Event event) throws Exception {
+        if(event.getName() == null){
+            throw new InvalidInputException("Id cannot be null");
+        }
         Event newEvent = null;
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
-        try {
-            newEvent = eventService.save(event);
-        } catch (Exception e) {
-            return new ResponseEntity(headers, HttpStatus.INTERNAL_SERVER_ERROR);
-        }
```

Comment:
See my larger comment where I explain why I removed all of the duplicate try catch blocks in your code. Ultimately there is an easier way to return these failed ResponseEntity objects.